types/events_test.go

```diff
@@ -68,6 +68,21 @@ func (s *eventsTestSuite) TestEventManager() {
 	s.Require().Equal(em.Events(), events.AppendEvent(event))
 }
 
+func (s *eventsTestSuite) TestEventManagerTypedEvents() {
+	em := sdk.NewEventManager()
+
+	coin1 := sdk.NewCoin("fakedenom", sdk.NewInt(1999999))
+	coin2 := sdk.NewCoin("fakedenom2", sdk.NewInt(1999999))
+
+	s.Require().NoError(em.EmitTypedEvents(&coin1))
+	s.Require().NoError(em.EmitTypedEvent(&coin2))
```

Comment:
This didn't cross my mind while reading the ADR, but does now. It seems weird that we can send an event that is a Coin.

Maybe we should create some convention, like the type name should be `Event...` or `Msg...Response`?